fog/ingest/server/src/worker.rs

```diff
@@ -275,3 +288,73 @@ impl Drop for PeerCheckupWorker {
         }
     }
 }
+
+/// The report cache worker is a thread responsible for periodically calling
+/// update report cache. This is a separate thread so that it can be on a
+/// time-based schedule, so it will happen even if there are few blocks.
+pub struct ReportCacheWorker {
```

Comment:
this worker object is added for fog-ingest, and it is not exactly the same as the one in `mc_sgx_report_cache_untrusted`. 

It can't be because, while in consensus, the "enclave identity" bytes never change, in fog ingest, they do, because sometimes the ingress key changes, and then the bytes in the report have to change.

(1) Sometimes, the operator calls "new_keys"
(2) Sometimes, the active node tells us "hey this is the key"
(3) Sometimes, we reach out to a remote node and say "hey give me your key" (if the operator asks us too)

Before this PR, we would get an updated report, whenever we were about to take an action involving reports or attestation.

After this PR, we update the report
* once when the server starts
* every 18 hours according to background thread, same as consensus
* after actions 1, 2, 3 listed above
* before some miscellaneous actions: activating the server, syncing key from remote. these two things happen very rarely (with a human in the loop) so it should be fine, and keeping these prevents us having to refactor tests.

Reply:
Thank you for the explanation here!

Reply:
i guess there is one more case:
- if we ask for a report to publish, and it doesn't match the key we think we have, then we update the report and try again.

Reply:
Maybe instead of duplicating it we should change `mc_sgx_report_cache_untrusted` to accept a callback for getting the report data? Not something I feel strongly about, since in this case the implementation is very small.

Reply:
yeah -- i think another way is, we could make an additional constructor of that thing that takes `Arc<Mutex<ReportCache>>` so that other things can also trigger the update?

it will be easier to develop changes like this after fog is merged into mobilecoin

Reply:
related question...why did you make a new ReportCacheWorker instead of just using the existing ReportCacheThread?  I see that you based the ReportCacheWorker on a PeerCheckupWorker--what does a PeerCheckupWorker do?

Reply:
so what happened here actually was, I looked at what ReportCacheThread does, and I copied it here, but made it just call `controller.update_enclave_report_cache`, because I still want the Controller to be able to trigger update sometimes, and so I want the controller to own the cache.

The philosophy I'm trying to use here is, there's one object that establishes invariants, owns most of the state, and has entrypoints where functionality can be called, that is like, not threaded and potentially unit-testable. And that's the controller. And all the threads, grpc routes, worker threads, etc. are outside that object, and call into it when they need something done. So that thing does have to have mutexes, but it can be done in a thread-safe way, and it means that to figure out the high level behavior of the server, if there is anyway it can deadlock etc., we mostly just have to read `controller.rs`. There's some logic in the `worker.rs` related to timings and such that's relevant, but anything where they are like changing the state of the system or talking to the enclave is going through the controller at some point.